### ISSUE TITLE
Play Integration doesn't capture all spans in request

### ISSUE THREAD

**Issue by schmohlio** [DataDog/dd-trace-java#926]
Greetings,

We are using Play framework and gRPC with the datadog APM. One request to Play may make multiple requests to gRPC in a single HTTP request.

We've noticed that without any additional instrumentation, tracing only captures the 1st gRPC span within the Play request. A reference architecture of this issue can be found here: https://github.com/schmohlio/dd-grpc-play-demo

We've opened tickets with Datadog support and haven't found any resolution. To resolve this issue, we add our own instrumentation to play, something like this:
```
public class DatadogTracingFilter extends Filter {

    @Inject
    public DatadogTracingFilter(Materializer mat) {
        super(mat);
    }

    @Override
    public CompletionStage<Result> apply(
            Function<Http.RequestHeader, CompletionStage<Result>> next,
            Http.RequestHeader rh) {

        String actionMethod = "..."
        final Tracer tracer = GlobalTracer.get();
        final Scope scope = tracer.buildSpan("...")
                .ignoreActiveSpan()
                .withTag("method", actionMethod)
                .withTag("uri", rh.uri())
                .startActive(true);
        // also tested this...
        //if (scope instanceof TraceScope) {
        //    ((TraceScope) scope).setAsyncPropagation(true);
        //}


        return next.apply(rh)
                .handle((result, ex) -> {
                    if (scope != null) {
                        scope.close();
                    }
                    if (ex != null) {
                        return null;
                    }
                    return result;
                });
    }
}
```

This seems to fix the missing span issue. However, we noticed a memory leak when running this in our system. It seems spans and traces are never cleaned up (after 130 requests):
![image](https://user-images.githubusercontent.com/3815022/61601293-cc5bef80-ac02-11e9-8ddd-aeaf8c155a02.png)
Further exploration with VisualVm shows objects moving to Old Gen and thousands objects created monotonically under load.

Any ideas? We are running the latest datadog apm jar, as well as open tracing 0.31.0, per website documentation.

**Comment by labbati** [comment]
Hi @schmohlio and thanks for opening this issue. We are looking into this as a stability issue and this is having high priority in our list. Will update here as soon as possible.

**Comment by rogerzxu** [comment]
Commenting to add more visibility to this issue.

We're encountering similar resource constraints in regards to object creation, it greatly impacts our CPU especially on application startup. 

In regards to missing spans, we are also experiencing the same issue, but instead of gRPC, our downstream JDBC requests are not properly being associated with the root HTTP request.

**Comment by tylerbenson** [comment]
Closing this as we think it was fixed in #953.
